src/PaginationCursor.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSauce\EventSourcing;
+
+interface PaginationCursor
+{
+    public function toString(): string;
+
+    public static function fromString(string $cursor): static;
+
+    public function isAtStart(): bool;
```

Comment:
This interface is missing the `offset()` method that is called [here][1].

[1]: https://github.com/EventSaucePHP/EventSauce/pull/181/files#diff-3372b852a1c039cb65c399c121bf1aa211f0cc1b3dea2df919a469c904f9f39eR82

Reply:
The interface isn't missing it. Offset is a property of the OffsetCurser. There might be different types of cursers that have different implementations. 

This is why there is this union typehint and type check in the inMemoryRepository (as well as there will eventually be in the real implementations)